Created LED subsystem and commands

ananyakidambi opened the pull request:
- tested on 2023 Slink
- able to change colors of LED through smartdashboard dropdowns
- able to switch on and off

ananyakidambi committed:
Merge remote-tracking branch 'origin/main' into akidambi

ananyakidambi committed:
Changed button commands

ananyakidambi committed:
Fixed setpoint conversion code

ananyakidambi committed:
Updated Prototype Code with Setpoints

ananyakidambi committed:
Added Power and LED Subsystems

ananyakidambi committed:
Created LED subsystem and command

ananyakidambi committed:
Merge remote-tracking branch 'origin/main' into akidambi

hpark-24 commented on Comp2023-Imported/src/main/java/frc/robot/subsystems/LED.java:
```diff
@@ -57,15 +57,14 @@ public void periodic( )
-    // This method will be called once per scheduler run when in simulation
-  }
 
-  // Put methods for controlling this subsystem here. Call these from Commands.
```
I don't think we should be making any changes to Comp2023-Imported. Could we keep what was there originally?

jmullins16 replied:
Ditto with what Hannah said. We should back out changes made to Comp2023-imported from this PR.

ananyakidambi replied:
I added it back in since I don't know how to revert changes on one specific file

hpark-24 commented on Comp2024/src/main/java/frc/robot/Constants.java:
It looks like a lot was changed here. I'm not sure if you meant for this and most of the constants seem the same. Is it possible to just have your own specific changes here (i.e. LED)?

jmullins16 replied:
Hannah is correct. Something has happened with the Constants.java file. We should see only the changes you neede--and since the LED constants were already in this file, there should have been none. We'll work on this together next time.

hpark-24 commented on Comp2024/src/main/java/frc/robot/Robot.java:
I'm not sure about this, but I think we want to refrain from adding things to Robot.java. Maybe we could move these changes to robotContainer?

jmullins16 replied:
Correct again. The current main branch does not support talking to subsystems from Robot.java. Maybe a merge from main into your branch is needed?

hpark-24 commented on Comp2024/src/main/java/frc/robot/RobotContainer.java:
This is similar to my comment about Constants.java. It also looks like some old code is here. Did you merge with main?

hpark-24 commented on Prototype2024/src/main/java/frc/robot/ExampleSmartMotorController.java:
Same comment as Constants.java in Comp2024. It looks mostly the same; did you mean to do this?

hpark-24 commented on Prototype2024/src/main/java/frc/robot/Robot.java:
Same comment as ExampleSmartMotorController.java (reference that comment).

hpark-24 reviewed (commented):
Overall, your subsystems look pretty good! Make sure you merge with main and review my comments. I think there are merge conflicts: you can either do what I told you today (resolving the conflicts) or we can work through it together tomorrow!

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/LED.java:
```diff
@@ -0,0 +1,145 @@
+  public void periodic( )
+  {
+    // This method will be called once per scheduler run
+    setColor(m_ledChooser.getSelected( ));
```
While this does work, it causes this subsystem to grab the dashboard setting and push it to the CANdle every 20 msec. even though the user might never change it. Generally beware of putting things in periodic methods unless they are necessary. The correct behavior would require the user to hit a SmartDashboard button that called this line of code from a command.

jmullins16 commented on Comp2024/src/main/java/frc/robot/subsystems/LED.java:
```diff
@@ -0,0 +1,145 @@
+      setColor(color);
+  }
+
+  public void setLLColor(LEDColor color)
```
I'm not exactly sure what normal mode is for, but it's likely different for 2024 and this method would not be needed. Need to investigate this one.

jmullins16 reviewed (changes requested):
The code is pretty close, but there's something in the merge process that's causing your branch to show lots of extraneous changes that aren't really there. We should discuss the process to find out what's going on with this.

ananyakidambi committed:
Merge remote-tracking branch 'origin/main' into akidambi

ananyakidambi committed:
Merge remote-tracking branch 'origin/main' into akidambi

ananyakidambi committed:
Reverted changes in Robot.java and RobotContainer

ananyakidambi committed:
Update ExampleSmartMotorController.java

ananyakidambi committed:
Fixed LED periodic commands and removed Power

frc2135 committed:
Reverted Changes in Comp2023-Imported and Prototype

hpark-24 committed:
Update ExampleSmartMotorController.java

jmullins16 committed:
Update Robot.java

Removed commented out call to Power subsystem initialization. This would not be done in Robot.java, but it can be done in the robotContainer.initialize() method.

jmullins16 reviewed (approved):
Looks ready to merge.
